src/utils/display-observer.js

```diff
@@ -117,24 +170,94 @@ export class DisplayObserver {
 
   /** @override */
   dispose() {
-    this.observers_.forEach((observer) => observer.disconnect());
+    this.observers_.forEach((observer) => {
+      if (observer.io) {
+        observer.io.disconnect();
+      }
+    });
+    this.observers_.length = 0;
     this.visibilityUnlisten_();
     this.visibilityUnlisten_ = null;
     this.targetObserverCallbacks_.clear();
     this.targetObservations_.clear();
   }
 
+  /**
+   * @param {!Element} container
+   * @param {?Element=} opt_root
+   */
+  registerContainer(container, opt_root) {
+    const index = findObserverByContainer(this.observers_, container);
+    if (index != -1) {
+      return;
+    }
+
+    this.observers_.push({
+      container,
+      root: opt_root || container,
+      contains: (target) => containsNotSelf(container, target),
+      // Start with null as IntersectionObserver. Will be initialized when
+      // the container itself becomes displayed.
+      io: null,
+    });
+    this.observe(container, this.containerObserved_);
```

Comment:
There's no race condition because it's supposed to eventually recover. But I do agree this is not obvious here and no good reason why. I added the observation reset immediately on registration like you recommend. And I added tests for this situation specifically.